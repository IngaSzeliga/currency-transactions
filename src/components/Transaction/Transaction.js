import React from "react";
import PropTypes from "prop-types";
import Fab from "@material-ui/core/Fab";
import CachedIcon from "@material-ui/icons/Cached";
import DeleteIcon from "@material-ui/icons/Delete";
import { countryRegion, EUR_NAME, PLN_NAME } from "../../config";
import "./Transaction.scss";

const Transaction = props => {
  const { id, title, amount, convertedAmount, deleteTransaction } = props;
  const amountFormatted = amount.toLocaleString(countryRegion, {
    style: "currency",
    currency: EUR_NAME
  });
  const convertedAmountFormatted = convertedAmount.toLocaleString(
    countryRegion,
    {
      style: "currency",
      currency: PLN_NAME
    }
  );
  return (
    <div className="transaction-container">
      <div>
        <CachedIcon fontSize="large" />
      </div>
      <div className="transaction-container-title">{title}</div>
      <div className="transaction-container-amounts">
        <div className="transaction-container-amounts-amount">
          {amountFormatted}
        </div>
        <div className="transaction-container-amounts-converted-amount">
          {convertedAmountFormatted}
        </div>
      </div>
      <Fab
        color="secondary"
        aria-label="delete"
        size="small"
        className="transaction-container-delete-button"
        onClick={() => deleteTransaction(id)}
      >
        <DeleteIcon />
      </Fab>
    </div>
  );
};

Transaction.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  amount: PropTypes.number.isRequired,
  convertedAmount: PropTypes.number.isRequired,
  deleteTransaction: PropTypes.func.isRequired
};

export default React.memo(Transaction);
