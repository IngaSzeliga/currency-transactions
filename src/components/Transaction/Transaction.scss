@import "../../config/style.scss";
@import "../../config/sizes.scss";

.transaction-container {
  display: flex;
  margin: 20px 0;
  .transaction-container-title {
    margin: 0 20px;
    font-weight: 600;
    @media only screen and (max-width: $phone) {
      margin: 0 10px;
      font-size: 14px;
    }
  }
  .transaction-container-amounts {
    .transaction-container-amounts-amount {
      font-size: 18px;
      @media only screen and (max-width: $phone) {
        font-size: 15px;
      }
    }
    .transaction-container-amounts-converted-amount {
      margin-top: 5px;
      color: $light-gray;
      font-size: 13px;
      text-align: center;
      @media only screen and (max-width: $phone) {
        font-size: 10px;
      }
    }
  }
  .transaction-container-delete-button {
    margin-left: 30px;
    @media only screen and (max-width: $phone) {
      margin-left: 10px;
    }
  }
}
