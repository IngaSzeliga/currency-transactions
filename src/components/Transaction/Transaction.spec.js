import React from "React";
import { shallow, mount } from "enzyme";
import { Fab } from "@material-ui/core";
import Transaction from ".";

describe("Transaction component", () => {
  const component = shallow(
    <Transaction
      id="some-id"
      title="some-title"
      amount={12}
      convertedAmount="12,33"
      deleteTransaction={jest.fn()}
    />
  );

  it("has working component", () => {
    expect(component).toBeDefined();
  });

  it("expects component to never be changed", () => {
    expect(component).toMatchSnapshot();
  });

  it("expects to call deleteTransaction", () => {
    const mockFunction = jest.fn();
    const componentMounted = mount(
      <Transaction
        id="some-id"
        title="some-title"
        amount={12}
        convertedAmount="12,33"
        deleteTransaction={mockFunction}
      />
    );
    componentMounted
      .find(Fab)
      .first()
      .simulate("click");
    expect(mockFunction).toBeCalled();
  });
});
