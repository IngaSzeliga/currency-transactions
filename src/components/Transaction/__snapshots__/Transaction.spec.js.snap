// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component expects component to never be changed 1`] = `
<WithStyles(ForwardRef(Paper))
  className="transaction-container"
>
  <div>
    <CachedIcon
      fontSize="large"
    />
  </div>
  <div
    className="transaction-container-title"
  >
    some-title
  </div>
  <div
    className="transaction-container-amounts"
  >
    <div
      className="transaction-container-amounts-amount"
    >
      € 12.00
    </div>
    <div
      className="transaction-container-amounts-converted-amount"
    >
      12,33
    </div>
  </div>
  <WithStyles(ForwardRef(Fab))
    aria-label="delete"
    className="transaction-container-delete-button"
    color="secondary"
    onClick={[Function]}
    size="small"
  >
    <DeleteIcon />
  </WithStyles(ForwardRef(Fab))>
</WithStyles(ForwardRef(Paper))>
`;
